src/ImageRunModal.jsx

```diff
@@ -393,17 +419,43 @@ export class ImageRunModal extends React.Component {
 
                         for (const result of reply) {
                             if (result.status === "fulfilled") {
-                                imageResults = imageResults.concat(JSON.parse(result.value));
+                                try {
+                                    imageResults = imageResults.concat(JSON.parse(result.value));
+                                } catch (err) {
+                                    console.error(err);
```

Comment:
How can this fail? If the promise succeeds, it should not throw random garbage at you, but properly formatted JSON. If this failed because the user typed some garbage, then it should be reflected in the UI somehow (but then *not* through a `JSON.parse()` exception, but a semantic error)